tests/qinfo/test_fidelities.py

```diff
@@ -0,0 +1,417 @@
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Unit tests for differentiable quantum entropies.
+"""
+
+import pytest
+
+import pennylane as qml
+from pennylane import numpy as np
+
+pytestmark = pytest.mark.all_interfaces
+
+tf = pytest.importorskip("tensorflow", minversion="2.1")
+torch = pytest.importorskip("torch")
+jax = pytest.importorskip("jax")
+
+
+def expected_fidelity_rx_pauliz(param):
+    """
+    Return the analytical fidelity for the RX and PauliZ.
+    """
+    return (np.cos(param / 2)) ** 2
+
+
+def expected_grad_fidelity_rx_pauliz(param):
+    """
+    Return the analytical fidelity for the RX and PauliZ.
+    """
+    return -np.sin(param) / 2
+
+
+class TestFidelityQnode:
+    """Tests for Fidelity function between two QNodes ."""
+
+    devices = ["default.qubit", "default.mixed"]
+
+    @pytest.mark.parametrize("device", devices)
+    def test_not_same_number_wires(self, device):
+        """Test that wires must have the same length"""
+        dev = qml.device(device, wires=1)
+
+        @qml.qnode(dev)
+        def circuit0():
+            return qml.state()
+
+        @qml.qnode(dev)
+        def circuit1():
+            return qml.state()
+
+        with pytest.raises(
+            qml.QuantumFunctionError, match="The two states must have the same number of wires"
+        ):
+            qml.qinfo.fidelity(circuit0, circuit1, wires0=[0, 1], wires1=[0])()
+
+    @pytest.mark.parametrize("device", devices)
+    def test_fidelity_qnodes_rx(self, device):
+        """Test the fidelity between two Qnodes."""
+        dev = qml.device(device, wires=1)
+
+        @qml.qnode(dev)
+        def circuit0(x):
+            qml.RX(x, wires=0)
+            return qml.state()
+
+        @qml.qnode(dev)
+        def circuit1(y):
+            qml.RX(y, wires=0)
+            return qml.state()
+
+        fid = qml.qinfo.fidelity(circuit0, circuit1, wires0=[0], wires1=[0])((0.1), (0.1))
+        assert qml.math.allclose(fid, 1.0)
+
+    @pytest.mark.parametrize("device", devices)
+    def test_fidelity_qnodes_rxrz_ry(self, device):
+        """Test the fidelity between two Qnodes."""
+        dev = qml.device(device, wires=1)
+
+        @qml.qnode(dev)
+        def circuit0(x, y):
+            qml.RX(x, wires=0)
+            qml.RY(y, wires=0)
+            return qml.state()
+
+        @qml.qnode(dev)
+        def circuit1(y):
+            qml.RY(y, wires=0)
+            return qml.state()
+
+        fid = qml.qinfo.fidelity(circuit0, circuit1, wires0=[0], wires1=[0])((0.0, 0.2), (0.2))
+        assert qml.math.allclose(fid, 1.0)
+
+    @pytest.mark.parametrize("device", devices)
+    def test_fidelity_qnodes_rx_empty(self, device):
```

Comment:
This test name is repeated 2 more times afterwards and the next case seems to be identical.